### PR TITLE
Use color picker white point to switch to white mode.

### PR DESCRIPTION
**Pull request by ultratoto14**
Fix also unwanted DP request that can generate unneeded log entry in DEBUG mode.

### CODE REVIEW

**Comment by FeikoJoosten** [comment]
@ultratoto14 seems to function as intended. However, did find another bug. If you change the brightness of the lamp, the previous color value is taken into account instead of the last assigned mode (white/color).

**Comment by ultratoto14** [comment]
@FeikoJoosten can you describe a little more the steps for your brightness issue ?

**Comment by FeikoJoosten** [comment]
@ultratoto14 I have the lamp configured as followed.
image

With the setup from the screenshot. The lamp is set to white with a low brightness. 
Adjusting the brightness changes the lamp to a blue color and allows me to change the brightness.
Adjusting the color temperature back to the desired color, reverts the brightness to the low brightness again (no clue what the brightness value is).

**Comment by ultratoto14** [comment]
On the screenshot, the white point is not centered, it means HA thinks your device is in color mode. 
If you use this PR, close the window and reopen it, the color picker should be really centered, if it's not the case, the white status is not reflected in HA. Do you use the tuya app at the same time ?

**Comment by ultratoto14** [comment]
Can you enable the [debugging](https://github.com/rospogrigio/localtuya#debugging), do the steps and post here the log file ?
It could also help to point the time for each step you are doing.

**Comment by FeikoJoosten** [comment]
Per your request here are the logs when changing the light settings.
```2020-11-23 15:30:00 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Sending command set (device type: type_0a)
2020-11-23 15:30:00 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Send payload: b'{"devId":"80157022e09806021245","uid":"80157022e09806021245","t":"1606145400","dps":{"1":true,"5":"727272002d0075","2":"colour"}}'
2020-11-23 15:30:00 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Waiting for sequence number 19
2020-11-23 15:30:00 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Dispatching message TuyaMessage(seqno=19, cmd=7, retcode=0, payload=b'', crc=1902033541)
2020-11-23 15:30:00 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Dispatching sequence number 19
2020-11-23 15:30:00 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Decrypted payload: {}
2020-11-23 15:30:01 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Sending command set (device type: type_0a)
2020-11-23 15:30:01 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Send payload: b'{"devId":"80157022e09806021245","uid":"80157022e09806021245","t":"1606145401","dps":{"1":true,"2":"white","3":3,"4":141}}'
2020-11-23 15:30:01 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Waiting for sequence number 20
2020-11-23 15:30:02 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Dispatching message TuyaMessage(seqno=20, cmd=7, retcode=0, payload=b'', crc=204324061)
2020-11-23 15:30:02 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Dispatching sequence number 20
2020-11-23 15:30:02 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Decrypted payload: {}
2020-11-23 15:30:03 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Sending command heartbeat (device type: type_0a)
2020-11-23 15:30:03 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Send payload: b'{}'
2020-11-23 15:30:03 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Waiting for sequence number -100
2020-11-23 15:30:03 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Dispatching message TuyaMessage(seqno=0, cmd=9, retcode=0, payload=b'', crc=2958142211)
2020-11-23 15:30:03 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Got heartbeat response
2020-11-23 15:30:03 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Decrypted payload: {}``` 

Full log: [home-assistant.log](https://github.com/rospogrigio/localtuya/files/5584008/home-assistant.log)

**Comment by ultratoto14** [comment]
Hi @FeikoJoosten thanks for the logs. What we can see is that the HA light does not receive any data from the device, all the decrypted payload are empty.
I also see in the sent commands that your brightness is set to 3 (**"3":3**):
```
2020-11-23 15:30:01 DEBUG (MainThread) [custom_components.localtuya.pytuya] [801...245] Send payload: b'{"devId":"80157022e09806021245","uid":"80157022e09806021245","t":"1606145401","dps":{"1":true,"2":"white","3":3,"4":141}}'
```
Can you set a min value of 25 as you showed in your [post](https://github.com/rospogrigio/localtuya/issues/143#issuecomment-732075549) and check again ?

**Comment by FeikoJoosten** [comment]
The issue still occurs if I set the brightness min to `25`. However, I think there might be something else going on as well, possibly causing this issue. The lamp seems to be unaware of it's actual state. For example, turning off the lamp in HA causes HA to turn it back on in lovelace. (The lamp does stay off though). 

Included updated log from the moment I've set the min brightness to `25`. [home-assistant.log](https://github.com/rospogrigio/localtuya/files/5584307/home-assistant.log)

**Comment by ultratoto14** [comment]
I had this issue before, when you set a value out of the range, the device may fail. Can you try to physically switch on and off the device ?

**Comment by FeikoJoosten** [comment]
Hmm and now I have a feeling I might need to reinstall LocalTuya. 
```
Platform localtuya does not generate unique IDs. ID local_80157022e09806021245_1 already exists - ignoring light.kitchen_light
Home Assistant has started!```

**Comment by ultratoto14** [comment]
You can first try to remove the entity form the integration page and restart HA first.

@postlund, this error mentioned by FeikoJoosten, also occurred sometimes on my setup, maybe it's linked to YAML device configuration migration. Are you aware of that one ? Is it problematic or an expected behavior ?

**Comment by FeikoJoosten** [comment]
Okay, readding the light and with the change from this PR resolves all of my issues. The light seems to function properly!

**Comment by postlund** [comment]
@ultratoto14 I think I've seen it in some other issue before, I don't know what has happened but I sure do want know. If you can manage to replicate, I would like to know how and also see the config entries created by the integration.

File: custom_components/localtuya/light.py [hunk 1]
```diff
@@ -366,10 +370,10 @@ def status_updated(self):
         self._state = self.dps(self._dp_id)
         supported = self.supported_features
         self._effect = None
-        if supported & SUPPORT_BRIGHTNESS:
+        if supported & SUPPORT_BRIGHTNESS and self.has_config(CONF_BRIGHTNESS):
```
Comment (postlund) [inline custom_components/localtuya/light.py:373]: `SUPPORT_BRIGHTNESS` is set if config exists for brightness, so this change is redundant (see `supported_features` and you'll understand what I mean).

File: custom_components/localtuya/light.py [hunk 2]
```diff
@@ -366,10 +370,10 @@ def status_updated(self):
...
             self._brightness = self.dps_conf(CONF_BRIGHTNESS)
 
-        if supported & SUPPORT_COLOR:
+        if supported & SUPPORT_COLOR and self.has_config(CONF_COLOR):
```
Comment (postlund) [inline custom_components/localtuya/light.py:376]: Same as above.

**Review by postlund — CHANGES_REQUESTED** [review]

File: custom_components/localtuya/light.py (hunk 1 above)
Comment (ultratoto14) [reply custom_components/localtuya/light.py:373]: This is not the only way, you can have RGB devices without white control, in that case support brightness is activated but via the setting in the color field.

Comment (postlund) [reply custom_components/localtuya/light.py:373]: Oh, right, I missed that. Carry on then.

**Review by postlund — APPROVED** [review]

File: custom_components/localtuya/light.py (hunk 2 above)
Comment (ultratoto14) [reply custom_components/localtuya/light.py]: Removed overzealous check. Don't know why I did that

**Comment by postlund** [comment]
Sorry 'bout the accidental close, I'm on mobile and the UI jumped all over the place and I was too fast...

**Comment by ultratoto14** [comment]
No problem.

**Pull request merged**